### PR TITLE
AppleHV - make gz ops sparse

### PR DESCRIPTION
**Pull request by baude**
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by rhatdan** [comment]
@giuseppe PTAL

File: pkg/machine/compression/copy.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+func (w *sparseWriter) Write(p []byte) (n int, err error) {
+	if isEmptyChunk(p) {
+		offset, err := w.writer.Seek(int64(len(p)), io.SeekCurrent)
+		if err != nil {
+			w.lastChunkSparse = false
+			return 0, err
+		}
+		_ = offset
+		w.lastChunkSparse = true
+		return len(p), nil
```
Comment (giuseppe) [inline pkg/machine/compression/copy.go:101]: should we handle each byte separately so we can detect sequences of zeros that are within a single frame?

For example, if the Write method is called with a buffer consisting of a 2MB sequence of zeros and the last byte is not equal to zero, wouldn't the current code fail to detect it?

Comment (baude) [reply pkg/machine/compression/copy.go:101]: the code appears to work and has been used in crc for some time; however, we fully intend to remove the copied file and actually vendor the code then.  So, if we have suggestions, we could take them to crc-org/crc/pkg/os ?

Comment (cfergeau) [reply pkg/machine/compression/copy.go:101]: crc uses this only through `CopySparse` which uses `CopyBuffer` to ensure `sparseWriter.Write()` is called with 4096 bytes chunks.
2MB sequence of zeros and non-zero last byte will indeed not be detected appropriately as being sparse.
This code could definitely be improved by iterating over the write buffer without relying on the calling code having 4k granularity. 64, 512 or 4k granularity for sparse detection might be less heavyweight than doing it byte by byte from a performance point of view.

File: pkg/machine/applehv/machine.go [hunk 2]
```diff
@@ -101,15 +100,7 @@ func checkProcessRunning(processName string, pid int) error {
...
-	// seems like os.truncate() is not very performant with really large files
-	// so exec'ing out to the command truncate
-	size := fmt.Sprintf("%dG", newSize)
-	c := exec.Command("truncate", "-s", size, mc.ImagePath.GetPath())
-	if logrus.IsLevelEnabled(logrus.DebugLevel) {
-		c.Stderr = os.Stderr
-		c.Stdout = os.Stdout
-	}
-	return c.Run()
+	return os.Truncate(mc.ImagePath.GetPath(), int64(newSize.ToBytes()))
```
Comment (cfergeau) [inline pkg/machine/applehv/machine.go:103]: I'm all for it, but I'd split that to its own commit. At some point, we can add a vfkit helper to do this even if it's a one liner.
